### ISSUE TITLE
Unexplainable image resizing artifacts

### ISSUE THREAD

**Issue by Arcaiyn** [Arclight-Team/Arclight#7]
This issue appears when applying scaling transformations (through e.g. `Image::resize`).

Upon resizing, independent of whether an image gets upscaled or downscaled, weird artifacts start to appear around the whole image. It manifests itself as "one-pixel wide stripes" in both directions, equidistantly spaced apart.

The following code reproduces the issue:
```cpp
Image image;    //Initialization omitted

image.applyFilter<GrayscaleFilter>();
image.applyFilter<SepiaFilter>();
image.applyFilter<ContrastFilter>(1);
image.applyFilter<InversionFilter>();
image.resize(ImageScaling::Bilinear, 700);    //Downscaling
```

The most prominent pixel format for this artifact to appear seems to be `Pixel::RGB5`, implying that lower-resolution pixel formats are more prone to this issue than higher-resolution ones.
Whether nearest-neighbor is also affected is unclear; tests have shown that this is not the case.

Commit 22ad2a713a54cf85729af485dcad0d1aa2c50e3a is affected by this behaviour.

**Comment by Arcaiyn** [comment]
After further investigation, it appears to happen whenever the new pixel's off-grid coordinates' fractional part becomes zero.
Logging output has proven this relationship:

(Exactly 20 stripes appear on the x axis)
```
[I: ] 17: 64.000000 1.000000 63 64
[I: ] 52: 192.000000 1.000000 191 192
[I: ] 87: 320.000000 1.000000 319 320
[I: ] 122: 448.000000 1.000000 447 448
[I: ] 157: 576.000000 1.000000 575 576
[I: ] 192: 704.000000 1.000000 703 704
[I: ] 227: 832.000000 1.000000 831 832
[I: ] 262: 960.000000 1.000000 959 960
[I: ] 297: 1088.000000 1.000000 1087 1088
[I: ] 332: 1216.000000 1.000000 1215 1216
[I: ] 367: 1344.000000 1.000000 1343 1344
[I: ] 402: 1472.000000 1.000000 1471 1472
[I: ] 437: 1600.000000 1.000000 1599 1600
[I: ] 472: 1728.000000 1.000000 1727 1728
[I: ] 507: 1856.000000 1.000000 1855 1856
[I: ] 542: 1984.000000 1.000000 1983 1984
[I: ] 577: 2112.000000 1.000000 2111 2112
[I: ] 612: 2240.000000 1.000000 2239 2240
[I: ] 647: 2368.000000 1.000000 2367 2368
[I: ] 682: 2496.000000 1.000000 2495 2496
```

Although no direct issue results from this observation, this might kick in a chain reaction of precision issues when calculating the interpolated color.

**Comment by Arcaiyn** [comment]
Pixel quantization during interpolation was the issue. Resolved with adace9b88beba5051ab163bece2d2312be21635f.
